config/sources/families/include/crust_firmware.inc

```diff
@@ -8,14 +8,16 @@
 #
 declare -g CRUST_TARGET_MAP="scp;;build/scp/scp.bin"
 
-[[ -n "${CRUSTCONFIG}" && -z $CRUSTSOURCE ]] && CRUSTSOURCE='https://github.com/crust-firmware/crust'
-[[ -n "${CRUSTCONFIG}" && -z $CRUSTDIR ]] && CRUSTDIR='crust-sunxi-mainline'
-[[ -n "${CRUSTCONFIG}" && -z $CRUSTBRANCH ]] && CRUSTBRANCH='commit:c308a504853e7fdb47169796c9a832796410ece8'
-[[ -n "${CRUSTCONFIG}" && -z $CRUST_USE_GCC ]] && CRUST_USE_GCC='> 9.1.0'
-[[ -n "${CRUSTCONFIG}" && -z $CRUST_COMPILER ]] && CRUST_COMPILER='or1k-elf-'
+if [[ -n "${CRUSTCONFIG}" ]]; then
+    [[ -z $CRUSTSOURCE ]] && CRUSTSOURCE='https://github.com/crust-firmware/crust'
```

Comment:
I think we will stil fail in here.

Please use invert logic like this:
```
[[ -n "$CRUSTSOURCE" ]] || CRUSTSOURCE='https://github.com/crust-firmware/crust'
```

Reply:
Could you please share the case where it fails? I was able to build images for boards which have crust enabled and also for boards that don't have crust enabled successfully. Hence asking for a case to test

Reply:
We have enabled `errexit`. So if `CRUSTSOURCE` is set, `[ -z $CRUSTSOURCE ]]` will return false. It is the result of this line, due to `errexit`, bash will trap ERR.

Reply:
Did you actually try building with CRUSTSOURCE set or are just speculating?

Because I did tried it and it didn't fail

Reply:
Sorry @hzyitc I don't follow either. Shortcircuits are ok with `errexit` if they're not the last statement in a function/sourced file. The last statement here is not a short-circuit anymore.
(I would have added `return 0` at the just like so many other places, it's simpler, but this should work)

Reply:
Sorry, my fault. It's work.

Why do I remember I have met this issue in the past ?

> Exit immediately if a pipeline (see [Pipelines](https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Pipelines)), which may consist of a single simple command (see [Simple Commands](https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Simple-Commands)), a list (see [Lists of Commands](https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Lists)), or a compound command (see [Compound Commands](https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Compound-Commands)) returns a non-zero status. The shell does not exit if the command that fails is part of the command list immediately following a while or until keyword, part of the test in an if statement, part of any command executed in a && or || list except the command following the final && or ||, any command in a pipeline but the last, or if the command’s return status is being inverted with !. If a compound command other than a subshell returns a non-zero status because a command failed while -e was being ignored, the shell does not exit. A trap on ERR, if set, is executed before the shell exits.
> 
> This option applies to the shell environment and each subshell environment separately (see [Command Execution Environment](https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Command-Execution-Environment)), and may cause subshells to exit before executing all the commands in the subshell.
> 
> If a compound command or shell function executes in a context where -e is being ignored, none of the commands executed within the compound command or function body will be affected by the -e setting, even if -e is set and a command returns a failure status. If a compound command or shell function sets -e while executing in a context where -e is ignored, that setting will not have any effect until the compound command or the command containing the function call completes.

Reply:
Well I'm totally traumatized by short-circuits in general, and short-circuits-as-last-statement and these days rather write the full `if...fi` in almost every case... and do `return 0` in functions even if not needed.... damn bash.